include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -5706,6 +5706,38 @@ struct OrtApi {
    */
   ORT_API2_STATUS(Graph_GetParentNode, _In_ const OrtGraph* graph, _Outptr_result_maybenull_ const OrtNode** node);
 
+  /** \brief Create a "sub-graph" from a subset of nodes in an OrtGraph.
+   *
+   * A "sub-graph" refers to a graph formed from a subset of nodes within the current OrtGraph and is not a
+   * nested subgraph. In contrast, a subgraph typically refers to a nested subgraph contained within a
+   * control flow node. If you are looking to retrieve nested subgraphs from a control flow node, use the
+   * Node_GetSubgraphs API.
+   * 
+   * Note:
+   * 'to_create_standalone_sub_graph' needs to be true to use 'copy_in_memory_initializer'.
+   * Regarding how initializers should be handled when constructing a new graph, in some cases,
+   * initializers that refer to a memory location in OrtValue can not be handled by some hardware backends
+   * (unlike those that are on disk). This prevents us from sharing the data and we have to make a copy here.
+   * In that case, set copy_in_memory_initializer to true.
+   *
+   * \param[in] src_graph The source OrtGraph instance.
+   * \param[in] nodes A subset of the nodes/OrtNodes in 'graph'.
+   * \param[in] num_nodes Number of nodes.
+   * \param[in] to_create_standalone_sub_graph If it's false, this API constructs a sub-graph as a OrtGraph instance
```

Comment:
Just curious, is there currently a need in TRT EP to create a completely standalone `OrtGraph` that lives outside of the calls to `GetCapability` or `Compile`?

Reply:
No, TRT EP only needs the `OrtGraph` that lives within the calls to `GetCapability `and `Compile`.
Not sure about other EPs, but I assume they don't need the OrtGraph to be completely standalone either.

Even though standalone OrtGraph is not needed in current EPs, i'm thinking do we want to keep the "create_standalone_ortgraph" in the API signature for future need?